spec.html

```diff
@@ -730,7 +730,6 @@
     <p>A step may specify the iterative application of its substeps.</p>
     <p>A step that begins with &ldquo;<dfn id="assert">Assert</dfn>:&rdquo; asserts an invariant condition of its algorithm. Such assertions are used to make explicit algorithmic invariants that would otherwise be implicit. Such assertions add no additional semantic requirements and hence need not be checked by an implementation. They are used simply to clarify algorithms.</p>
     <p>Algorithm steps may declare named aliases for any value using the form &ldquo;Let _x_ be _someValue_&rdquo;. These aliases are reference-like in that both _x_ and _someValue_ refer to the same underlying data and modifications to either are visible to both. Algorithm steps that want to avoid this reference-like behaviour should explicitly make a copy of the right-hand side: &ldquo;Let _x_ be a copy of _someValue_&rdquo; creates a shallow copy of _someValue_.</p>
-    <p>Once declared, aliases may be referenced in any subsequent step or substep thereof, but they may not be referenced from steps at a higher level. Aliases may be modified using the form &ldquo;Set _x_ to _someOtherValue_&rdquo;.</p>
```

Comment:
I don't think a total revert here is necessary - I still like most of this change. Can we just remove the step scoping requirement (i.e. should never refer to a subsequent step of any depth but may refer to a previous step at any depth)?

Reply:
So we keep:
> Aliases may be modified using the form &ldquo;Set _x_ to _someOtherValue_&rdquo;.

And if necessary, maybe something along the lines of:
> Aliases cannot be used before their declaration.

Reply:
Yeah, that's my thinking. Want to update this? Or I can exclude this commit too.